packages/flutter_tools/lib/src/web/file_generators/flutter_js.dart

```diff
@@ -20,6 +20,21 @@ _flutter.loader = null;
 
 (function () {
   "use strict";
+
+  const baseUri = ensureTrailingSlash(getBaseURI());
+
+  function getBaseURI() {
+    const base = document.querySelector("base");
+    return (base && base.getAttribute("href")) || "";
+  }
```

Comment:
Would it be possible to use `document.baseURI` to retrieve this instead of the selector?

https://developer.mozilla.org/en-US/docs/Web/API/Node/baseURI

(It also seems to guarantee the trailing slash?)

Reply:
That's how I [did it initially](https://github.com/flutter/flutter/pull/117691/commits/4a94af02ce10cb02a3bc7c25d29b54e8a655d3e3#diff-818be2e3cd2b55c69070b0fca9216af012cc8e42ba30f02bfcad697dcdb3c71cR13) but that was causing mysterious test failures (the test harness launches Chrome and then waits forever but Chrome never connects back). Turned out that was because some tests (specifically integration tests), don't have a `<base href="...">` and the test is served on a sub-path that's not the root, but `document.baseURI` points to `/` so it makes all the paths become absolute paths.

I had to remove `document.baseURI` and instead get the `href` from the `<base>` element ([here](https://github.com/flutter/flutter/pull/117691/commits/f0e0e35bf1742e69b20f7995c17493e32c81557d)). If a `<base>` element isn't available, we default to an empty string, which has no effect on the request paths, so everything works correctly.

Reply:
That's so odd, I'd expect that if the base is not present, `document.baseURI` defaults to whatever sub-path the page is served from (in an absolute URL), rather than "/" :/

Reply:
> That's so odd, I'd expect that if the base is not present, `document.baseURI` defaults to whatever sub-path the page is served from (in an absolute URL), rather than "/" :/

Actually, I think you are right. I may have misremembered. Either way, the default `document.baseURI` wasn't working, so I had to parse manually and default to empty string to make it work.